runner/ollamarunner/runner.go

```diff
@@ -256,6 +257,11 @@ type Server struct {
 	// multimodalHash generates hashes for comparing equality
 	// of non-text data
 	multimodalHash maphash.Hash
+
+	// llamaModel is used to get the vocab of the model
+	// TODO (jmorganca): remove this once we no longer
+	// rely on llama.cpp's grammar parsing
+	vocab *llama.Vocab
```

Comment:
This should be handled in the `sample` package.. working on it..

Reply:
yeah was just thinking about that - cool!

Reply:
I've kept it here since it ended up being easier to continue using the model loading code for now. We could probably only do this on first constrained generation instead of on startup